### PR TITLE
Use helpers to move gzip & sha from makefile to bazel

### PR DESCRIPTION
**Pull request by justinsb**
We can move the gzip & sha steps from the makefile to bazel.  This not only
simplifies the makefile, it should enable more parallelism and better
caching.

(Builds on #7702)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -806,8 +805,8 @@ bazel-version-dist: bazel-crossbuild-nodeup bazel-crossbuild-kops bazel-protokub
 	mkdir -p ${BAZELUPLOAD}/kops/${VERSION}/images/
 	mkdir -p ${BAZELUPLOAD}/utils/${VERSION}/linux/amd64/
 	cp bazel-bin/cmd/nodeup/linux_amd64_pure_stripped/nodeup ${BAZELUPLOAD}/kops/${VERSION}/linux/amd64/nodeup
-	tools/sha1 ${BAZELUPLOAD}/kops/${VERSION}/linux/amd64/nodeup ${BAZELUPLOAD}/kops/${VERSION}/linux/amd64/nodeup.sha1
-	tools/sha256 ${BAZELUPLOAD}/kops/${VERSION}/linux/amd64/nodeup ${BAZELUPLOAD}/kops/${VERSION}/linux/amd64/nodeup.sha256
+	(${SHASUMCMD} ${BAZELUPLOAD}/kops/${VERSION}/linux/amd64/nodeup ${BAZELUPLOAD}/kops/${VERSION}/linux/amd64/nodeup.sha1
+	(${SHA256SUMCMD} ${BAZELUPLOAD}/kops/${VERSION}/linux/amd64/nodeup ${BAZELUPLOAD}/kops/${VERSION}/linux/amd64/nodeup.sha256
```
Comment (rifelpet) [inline Makefile]: there are unbalanced parentheses on these 2 lines

**Comment by justinsb** [comment]
Thanks for putting up with my failed rebases @rifelpet :-)

**Pull request merged**